presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/Partition.java

```diff
@@ -58,43 +55,40 @@ public Partition(
             @JsonProperty("eligibleToIgnore") boolean eligibleToIgnore,
             @JsonProperty("sealedPartition") boolean sealedPartition,
             @JsonProperty("createTime") int createTime,
-            @JsonProperty("lastDataCommitTime") long lastDataCommitTime)
+            @JsonProperty("lastDataCommitTime") long lastDataCommitTime,
+            @JsonProperty("lastAccessTime") int lastAccessTime)
     {
-        this.databaseName = requireNonNull(databaseName, "databaseName is null");
-        this.tableName = requireNonNull(tableName, "tableName is null");
-        this.values = ImmutableList.copyOf(requireNonNull(values, "values is null"));
+        super(values, databaseName, tableName, createTime, lastAccessTime, makeStorageDescriptor(tableName, columns, storage, new HiveColumnConverter()), parameters);
         this.storage = requireNonNull(storage, "storage is null");
         this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
-        this.parameters = ImmutableMap.copyOf(requireNonNull(parameters, "parameters is null"));
         this.partitionVersion = requireNonNull(partitionVersion, "partitionVersion is null");
         this.eligibleToIgnore = eligibleToIgnore;
         this.sealedPartition = sealedPartition;
-        this.createTime = createTime;
         this.lastDataCommitTime = lastDataCommitTime;
     }
 
     @JsonProperty
     public String getDatabaseName()
     {
-        return databaseName;
+        return super.getDbName();
```

Comment:
Can we evaluate if we really need @JsonProperty,  @JsonCreator for this Partition class. If not, we can get rid of these duplicate getters.